src/sysupdate/updatectl.c

```diff
@@ -0,0 +1,1226 @@
+        if (r < 0)
+                return r;
+
+        if (urlify_enabled() && v.changelog)
+                lnk = special_glyph(SPECIAL_GLYPH_EXTERNAL_LINK);
+        update = strjoin(empty_to_dash(v.version), " ",
+                         special_glyph(SPECIAL_GLYPH_ARROW_RIGHT), " ",
+                         v.version, strempty(lnk));
+        if (!update)
+                return log_oom();
+
+        r = table_add_many(table,
+                           TABLE_STRING, data->target_id,
+                           TABLE_STRING, update,
+                           TABLE_SET_URL, v.changelog);
+        if (r < 0)
+                return table_log_add_error(r);
+
+        return 0;
+}
+
+static int check_finished(sd_bus_message *reply, void *userdata, sd_bus_error *ret_error) {
+        _cleanup_(userdata_freep) Userdata *data = ASSERT_PTR(userdata);
+        const sd_bus_error *e;
+        const char *new_version = NULL;
+        int r;
+
+        assert(reply);
+
+        e = sd_bus_message_get_error(reply);
+        if (e)
+                return log_bus_error(0, e, data->target_id, "call CheckNew");
+
+        r = sd_bus_message_read(reply, "s", &new_version);
+        if (r < 0)
+                return bus_log_parse_error(r);
+
+        if (isempty(new_version))
+                return 0;
+
+        r = sd_bus_call_method_async(data->bus, NULL, bus_sysupdate_mgr->destination, data->target_path, SYSUPDATE_TARGET_INTERFACE, "Describe",
+                                  check_describe_finished, data,
+                                  "sb", new_version, arg_offline);
+        if (r < 0)
+                return r;
+        TAKE_PTR(data);
+
+        return 0;
+}
+
+static int verb_check(int argc, char **argv, void *userdata) {
+        sd_bus *bus = ASSERT_PTR(userdata);
+        _cleanup_(table_unrefp) Table *table = NULL;
+        _cleanup_(sd_event_unrefp) sd_event *event = NULL;
+        _cleanup_strv_free_ char **targets = NULL;
+        _cleanup_strv_free_ char **target_paths = NULL;
+        size_t n;
+        unsigned remaining = 0;
+        int r;
+
+        r = ensure_targets(bus, argv + 1, &targets);
+        if (r < 0)
+                return r;
+
+        r = parse_targets(targets, &n, &target_paths, /* ret_versions= */ NULL);
+        if (r < 0)
+                return r;
+
+        table = table_new("target", "update");
+        if (!table)
+                return log_oom();
+
+        (void) table_set_sort(table, 0);
+
+        r = sd_event_default(&event);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get event loop: %m");
+
+        r = sd_bus_attach_event(bus, event, 0);
+        if (r < 0)
+                return log_error_errno(r, "Failed to attach bus to event loop: %m");
+
+        r = sd_event_set_signal_exit(event, true);
+        if (r < 0)
+                return log_error_errno(r, "Failed to set up interrupt handler: %m");
+
+        for (size_t i = 0; i < n; i++) {
```

Comment:
FOREACH_ARRAY

Reply:
This one also is indexing two arrays, so I don't think I can use a FOREACH_ARRAY.

Reply:
Indeed. Again, `target_paths[i]` and `targets[i]`